internal/cortex/frontend/transport/handler.go

```diff
@@ -183,8 +174,13 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		queryString = f.parseRequestQueryString(r, buf)
 
 		// Check if caching is enabled.
-		if f.lruCache != nil {
-			f.updateFailedQueryCache(err, queryExpressionNormalized, queryExpressionRangeLength, r)
+		if f.failedQueryCache != nil {
+			success, message := f.failedQueryCache.UpdateFailedQueryCache(err, queryExpressionNormalized, queryExpressionRangeLength, f.failedQueryCache.LruCache)
+			if success {
+				level.Info(util_log.WithContext(r.Context(), f.log)).Log(message)
+			} else {
+				level.Error(util_log.WithContext(r.Context(), f.log)).Log(message)
+			}
```

Comment:
Move this piece of logic to utils.go.
```
if f.failedQueryCache != nil && if f.failedQueryCache.HitCache(...) {
   ...
}

Reply:
+1, this logic is a bit verbose, could we just log info or log debug?

Reply:
removed, and consolidated into a debug log